packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -140,7 +140,15 @@ abstract class Route<T> {
   ///
   /// If the [settings] are not provided, an empty [RouteSettings] object is
   /// used instead.
-  Route({ RouteSettings? settings }) : _settings = settings ?? const RouteSettings();
+  Route({ RouteSettings? settings }) : _settings = settings ?? const RouteSettings() {
+    if (kFlutterMemoryAllocationsEnabled) {
+      MemoryAllocations.instance.dispatchObjectCreated(
+        library: 'package:flutter/widgets.dart',
+        className: '$Route<$T>',
```

Comment:
I didn't notice this earlier, why would you need to provide a className if you already provide the object itself? Should the ObjectCreated uses the object.runtimeType if it needs the class name?

Reply:
runtimeType is different from type that owns instrumentation, because runTime type is can me descender. Does it help?

Reply:
thanks for explanation